src/NodaTime.Demo/IntervalDemo.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright 2019 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NUnit.Framework;
+
+namespace NodaTime.Demo
+{
+    public class IntervalDemo
+    {
+        [Test]
+        public void Construction()
+        {
+            Instant start = Instant.FromUtc(2019, 1, 1, 15, 25, 48);
+            Instant end = Instant.FromUtc(2019, 1, 1, 16, 25, 48);
+            Interval interval = Snippet.For(new Interval(start, end));
+            Assert.AreEqual(start, interval.Start);
+            Assert.AreEqual(end, interval.End);
+        }
+
+        [Test]
+        public void Construction_WithNoBeginning()
+        {
+            Instant end = Instant.FromUtc(2019, 1, 1, 16, 25, 48);
+            Interval interval = Snippet.For(new Interval(null, end));
+            Assert.False(interval.HasStart);
+            Assert.AreEqual(end, interval.End);
+        }
+
+        [Test]
+        public void Construction_WithNoEnd()
+        {
+            Instant start = Instant.FromUtc(2019, 1, 1, 16, 25, 48);
+            Interval interval = Snippet.For(new Interval(start, null));
+            Assert.False(interval.HasEnd);
+            Assert.AreEqual(start, interval.Start);
+        }
+
+        [Test]
+        public void Construction_EternalInterval()
```

Comment:
We only support a single snippet per member at the moment. So we need two snippets - one for the `Interval(Instant, Instant)` constructor, and one for `Interval(Instant?, Instant?)`. I'd suggest joining these three examples together just with comments.

Reply:
Ok, will do.